contracts/libraries/QuestClaimable.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import { IQuestFactory } from "../interfaces/IQuestFactory.sol";
+
+abstract contract QuestClaimable {
+    function getQuestFactoryContract() public view virtual returns (IQuestFactory);
+
+    function getQuestId() public view virtual returns (string memory);
+
+    function claim() external payable {
```

Comment:
How does this get handled when settled to L1 and how does the gas used work in this call?